google/cloud/aiplatform/models.py

```diff
@@ -748,20 +776,24 @@ def _deploy_call(
             raise ValueError(
                 "Both `accelerator_type` and `accelerator_count` should be specified or None."
             )
-        if bool(explanation_metadata) != bool(explanation_parameters):
-            raise ValueError(
-                "Both `explanation_metadata` and `explanation_parameters` should be specified or None."
-            )
+        generate_explanation = cls._validate_explanation_args(
```

Comment:
I think this validation should be moved to _validate_deployment_args so it fails early.